### PR TITLE
core: reject bad fields in x509 guard data

### PR DESCRIPTION
**Pull request by kr**

### CODE REVIEW

File: core/grants.go [hunk 1]
```diff
@@ -31,11 +31,19 @@ func (a *API) createGrant(ctx context.Context, x apiGrant) error {
...
+				return errors.WithDetail(httpjson.ErrBadRequest, "bad subject field "+k)
+			}
+		}
 	}
 
 	// NOTE: package json produces consistent serialization output,
 	// effectively an ad hoc canonical form. We rely on this to
 	// grant data for equality.
+	// TODO(kr): x509 subject field names are case-insensitive,
+	// so we should do our equality comparisons accordingly.
```
Comment (tessr) [inline core/grants.go]: Is this todo still relevant?

Comment (kr) [inline core/grants.go]: Yes I think so. Don't we currently use `bytes.Equal` to check if two GrantData fields are equal? Those fields are serialized from the user-supplied json directly, so there's no case-folding in there yet.

Comment (tessr) [inline core/grants.go]: oh, yes, yes we do.

**Comment by tessr** [comment]
LGTM

**Pull request merged**